Apps/WebClient/src/Server/Listeners/BannerListener.cs

```diff
@@ -74,40 +75,52 @@ public class BannerListener : BackgroundService
+            while (!stoppingToken.IsCancellationRequested)
             {
-                using var scope = this.services.CreateScope();
-                GatewayDbContext dbContext = scope.ServiceProvider.GetRequiredService<GatewayDbContext>();
-                this.CommunicationService = scope.ServiceProvider.GetRequiredService<ICommunicationService>();
-                NpgsqlConnection con = (NpgsqlConnection)dbContext.Database.GetDbConnection();
-                con.Open();
-                con.Notification += this.ReceiveEvent;
-                using NpgsqlCommand cmd = new NpgsqlCommand();
-                cmd.CommandText = @$"LISTEN ""{Channel}"";";
-                cmd.CommandType = CommandType.Text;
-                cmd.Connection = con;
-                cmd.ExecuteNonQuery();
+                attempts++;
+                this.logger.LogInformation($"Registering Channel Notification on channel {Channel}, attempts = {attempts}");
+                try
+                {
+                    using var scope = this.services.CreateScope();
+                    using GatewayDbContext dbContext = scope.ServiceProvider.GetRequiredService<GatewayDbContext>();
+                    this.CommunicationService = scope.ServiceProvider.GetRequiredService<ICommunicationService>();
+                    using NpgsqlConnection con = (NpgsqlConnection)dbContext.Database.GetDbConnection();
+                    con.Open();
+                    con.Notification += this.ReceiveEvent;
+                    using NpgsqlCommand cmd = new NpgsqlCommand();
+                    cmd.CommandText = @$"LISTEN ""{Channel}"";";
+                    cmd.CommandType = CommandType.Text;
+                    cmd.Connection = con;
+                    cmd.ExecuteNonQuery();
 
-                while (!stoppingToken.IsCancellationRequested)
+                    while (!stoppingToken.IsCancellationRequested)
+                    {
+                        // Wait for the event
+                        await con.WaitAsync(stoppingToken).ConfigureAwait(true);
+                    }
+
+                    con.Close();
+                }
+                catch (NpgsqlException e)
                 {
-                    // Wait for the event
-                    await con.WaitAsync(stoppingToken).ConfigureAwait(true);
+                    this.logger.LogError($"DB Error encountered in WaitChannelNotification: {Channel}\n{e.ToString()}");
+                    if (!stoppingToken.IsCancellationRequested)
+                    {
+                        Thread.Sleep(SleepDuration);
+                    }
                 }
 
-                this.logger.LogInformation($"Exiting WaitChannelNotification: {Channel}");
-            }
-            catch (NpgsqlException e)
-            {
-                this.logger.LogError($"DB Error encountered in WaitChannelNotification: {Channel}\n{e.ToString()}");
+                this.logger.LogWarning($"DBChangeListener on {Channel} exiting...");
```

Comment:
any reasons why we log warning when exception occurs here?

Reply:
I was thinking that the outer loop is recovering the exception which means its no longer an error state but rather a warning which we should investigate.